main.go

```diff
@@ -182,53 +190,69 @@ func main() {
+	log.Infof("Export HTML results:")
 	fmt.Println()
 
-	reports, err := getArtifacts(gradleProject, started, config.ReportPathPattern, true, true)
+	reports, err := getArtifacts(gradleProject, started, config.HTMLResultDirPattern, true, true)
 	if err != nil {
 		failf("Failed to find reports, error: %v", err)
 	}
 
-	if err := exportArtifacts(deployDir, reports); err != nil {
+	if err := exportArtifacts(config.DeployDir, reports); err != nil {
 		failf("Failed to export reports, error: %v", err)
 	}
 
 	fmt.Println()
-
-	log.Infof("Export results:")
+	log.Infof("Export XML results:")
 	fmt.Println()
 
-	results, err := getArtifacts(gradleProject, started, config.ResultPathPattern, true, true)
+	results, err := getArtifacts(gradleProject, started, config.XMLResultDirPattern, true, true)
 	if err != nil {
 		failf("Failed to find results, error: %v", err)
 	}
 
-	if err := exportArtifacts(deployDir, results); err != nil {
+	if err := exportArtifacts(config.DeployDir, results); err != nil {
 		failf("Failed to export results, error: %v", err)
 	}
 
-	log.Infof("Export test results for test addon:")
-	fmt.Println()
+	if config.TestResultDir != "" {
+		// Test Addon is turned on
+		fmt.Println()
+		log.Infof("Export XML results for test addon:")
+		fmt.Println()
 
-	resultXMLs, err := getArtifacts(gradleProject, started, resultArtifactPathPattern, false, false)
-	if err != nil {
-		log.Warnf("Failed to find test result XMLs, error: %s", err)
-	} else {
-		if baseDir := os.Getenv("BITRISE_TEST_RESULT_DIR"); baseDir != "" {
+		lastOtherDirIdx := -1
+		xmlResultFilePattern := config.XMLResultDirPattern
+		if !strings.HasSuffix(xmlResultFilePattern, "*.xml") {
+			xmlResultFilePattern += "*.xml"
+		}
+
+		resultXMLs, err := getArtifacts(gradleProject, started, xmlResultFilePattern, false, false)
+		if err != nil {
+			log.Warnf("Failed to find test XML test results, error: %s", err)
+		} else {
 			for _, artifact := range resultXMLs {
-				uniqueDir, err := getUniqueDir(artifact.Path)
-				if err != nil {
-					log.Warnf("Failed to export test results for test addon: cannot get export directory for artifact (%s): %s", err)
-					continue
+				dir := getExportDir(artifact.Path)
+
+				if dir == OtherDirName {
+					// start indexing other dir name, to avoid overrideing it
+					// e.g.: other, other-1, other-2
+					lastOtherDirIdx++
+					if lastOtherDirIdx > 0 {
+						dir = dir + "-" + strconv.Itoa(lastOtherDirIdx)
+					}
 				}
 
-				if err := testaddon.ExportArtifact(artifact.Path, baseDir, uniqueDir); err != nil {
+				if err := testaddon.ExportArtifact(artifact.Path, config.TestResultDir, dir); err != nil {
 					log.Warnf("Failed to export test results for test addon: %s", err)
+				} else {
+					src := artifact.Path
+					if rel, err := workDirRel(artifact.Path); err == nil {
+						src = "./" + rel
+					}
+					log.Printf("  Export [%s => %s]", src, filepath.Join("$BITRISE_TEST_RESULT_DIR", dir, filepath.Base(artifact.Path)))
```

Comment:
Why use the relative path in the log?

Reply:
The previous log only printed the artifact's name, which was not enough informative, full artifact path was too long, so I decided to print relative paths.